PYME/cluster/HTTPDataServer.py

```diff
@@ -679,6 +681,64 @@ def get_h5_part(self, path):
         except IOError:
             self.send_error(404, "File not found - %s, [%s]" % (self.path, path))
             return None
+
+    def get_tabular_part(self, path):
+        """
+
+        Parameters
+        ----------
+        path: str
+            path to an hdf or h5r file on the dataserver computer. Specific parts of the file can be accessed by
+            appending, e.g. /Metadata, for a resulting `path` of something like folder/test.h5r/Metadata. The 'part' of
+            the file returned defaults to FitResults if unspecified, and can additionally include a slice to take before
+            returning. In other words, a `path` of 'folder/test.h5r/FitResultsslice(0,100,None)' will return the first
+            hundred elements of the FitResults table in test.h5r.
+
+        Returns
+        -------
+        f: BytesIO
+            Requested part of the file encoded as bytes
+
+        """
+        from PYME.IO.tabular import NestedRecArraySource
+        from PYME.IO import h5rFile
+        ext = '.h5r' if '.h5r' in path else '.hdf'
+        try:
+            filename, part = path.split(ext)
+            ctype = self.guess_type(filename + ext)
+            part = part.lstrip('/').lstrip('\\')
+
+            with h5rFile.openH5R(filename + ext) as h5f:
+                if part == 'Metadata':
+                    mdh = h5f.mdh.to_JSON()
+                    f, length = self._string_to_file(mdh)
+                elif part == 'Events':
+                    # events = h5f.events
+                    #  fixme / make events class with to_JSON method
+                    self.send_error(400, 'serving events from tabular files not yet supported')
+                    return
+                else:
+                    if 'slice' in part:
+                        # convert the slice back from str, avoiding eval for security
+                        part, return_slice = part.strip(')').split('slice(')
+                        return_slice = slice(*(None if s == 'None' else int(s) for s in return_slice.split(',')))
+                    else:
+                        return_slice = slice(None)
+                        if part == '':
+                            part = 'FitResults'
```

Comment:
I don't think we need to have any special case logic for 'FitResults' on the server side